File: app/src/main/java/com/juawapps/newsspread/ui/news/ReadMoreHandler.java
package com.juawapps.newsspread.ui.news;

import android.view.View;

import com.juawapps.newsspread.web.CustomTabHelper;

/**
 * Handler class to manage news items clicks.
 */
@SuppressWarnings("WeakerAccess")
public class ReadMoreHandler {

    private final CustomTabHelper mCustomTabHelper;

    public ReadMoreHandler(CustomTabHelper customTabHelper) {
        mCustomTabHelper = customTabHelper;
    }

    public void onReadMoreClick(String url, View view) {
        mCustomTabHelper.openNewTab(url, view.getContext());
    }
}

File: app/src/main/java/com/juawapps/newsspread/ui/news/ReadMoreHandler.kt
package com.juawapps.newsspread.ui.news

import android.view.View

import com.juawapps.newsspread.web.CustomTabHelper

/**
 * Handler class to manage news items clicks.
 */
class ReadMoreHandler(private val mCustomTabHelper: CustomTabHelper) {

    fun onReadMoreClick(url: String, view: View) {
        mCustomTabHelper.openNewTab(url, view.context)
    }
}
